accord-core/src/main/java/accord/impl/InMemoryCommandStore.java

```diff
@@ -774,7 +784,8 @@ public <T> T mapReduce(Seekables<?, ?> keysOrRanges, Ranges slice, TestKind test
                     if (!command.known().deps.hasProposedOrDecidedDeps())
                         return;
 
-                    if ((testDep == WITH) == !command.partialDeps().contains(depId))
+                    // TODO (required): ensure C* matches this behaviour
+                    if ((testDep == WITH) == !command.partialDeps().intersects(depId, rangeCommand.ranges))
```

Comment:
We are looking for transactions A that have B as a dependency. If B covers ranges [1..3] and A covers [2..3], but the command store only covers ranges [1..2], we could have A adopt B as a dependency on [3..3] only, and have that A intersects B on this command store but also that there is no dependency relation between them on the overlapping key range [2..2].

This may or may not actually cause problems in practice. But it is safer to ensure that we make proper independent decisions for each key range, as we don't actually have a dependency relation on these keys so if we recover B we should invalidate it since we know it did not reach a fast path decision.

Reply:
Hmm, that said, I need to think this through some more in particular with respect to how it interacts with our compression of the individual command store votes to a single node vote. I think this is fine still, but worth formalising a little.